openscm/parameter_views.py

```diff
@@ -128,21 +68,26 @@ def get(self) -> float:
         If the parameter has child parameters (aka ``_children`` is not empty),
         the returned value will be the sum of the values of all of the child
         parameters.
+
+        Raises
+        ------
+        ParameterEmptyError
+            Parameter is empty, i.e. has not yet been written to
         """
-        if self.is_empty:
-            return self._attempt_aggregate_child_data()
+        if self._parameter._children:
+            return sum(v.get() for v in self._child_data_views)
+        elif self.is_empty:
+            raise ParameterEmptyError
 
         return self._unit_converter.convert_from(self._parameter._data)
 
-    def _get_child_view(self, child_parameter: _Parameter) -> ParameterView:
-        return type(self)(child_parameter, self._unit_converter._target)
-
-    def _sum_child_data_views(self, child_data_views: list) -> float:
-        data = 0
-        for v in child_data_views:
-            data += v.get()
-
-        return data
+    def _get_child_data_views(self, parameter: _Parameter) -> Sequence["ScalarView"]:
+        if parameter._children:
+            return sum(
+                (self._get_child_data_views(p) for p in parameter._children.values()),
+                [],
+            )
+        return [ScalarView(parameter, self._unit_converter._target)]
```

Comment:
if there are no children, but we ask for children, shouldn't we throw an error rather than just return self? Or is this needed for the recursion to work?